File: Algorithm/Code_Materials/python_source/charpter17_greedy/level1/find_content_children.py
# coding=utf-8
class FindContentChildren:
    def findContentChildren(self, g, s):
        g.sort()
        s.sort()
        m, n = len(g), len(s)
        wei_kou = bing_gan = count = 0

        while wei_kou < m and bing_gan < n:
            while bing_gan < n and g[wei_kou] > s[bing_gan]:
                bing_gan += 1
            if bing_gan < n:
                count += 1
            wei_kou += 1
            bing_gan += 1

        return count


if __name__ == '__main__':
    g = [1, 2, 3]
    s = [1, 1]
    findContentChildren = FindContentChildren()
    print findContentChildren.findContentChildren(g, s)

File: Algorithm/Leetcode/Code_Materials/c_or_cpp/charpter16_slide/level2/find_content_children.cpp
#include <iostream>
#include<vector>
using namespace std;
 
 int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int m = g.size(), n = s.size();
        int count = 0;
        for (int i = 0, j = 0; i < m && j < n; i++, j++) {
            while (j < n && g[i] > s[j]) {
                j++;
            }
            if (j < n) {
                count++;
            }
        }
        return count;
    }
 

int main() {
    std::vector<int> g = {1,2,3};
    std::vector<int> s = {1,1};
    int count = findContentChildren(g, s);
    std::cout << "Number of content children: " << count << std::endl;
    return 0;
     
}